packages/headless/src/controllers/result-list/headless-result-list.test.ts

```diff
@@ -105,5 +110,18 @@ describe('ResultList', () => {
       expect(engine.logger.error).not.toHaveBeenCalled();
       done();
     });
+
+    it(`when calling fetchMoreResults while there are no more results available
+    is should not dispatch and log an info`, async (done) => {
+      engine.state.search.response.totalCountFiltered =
+        engine.state.search.response.results.length;
+      await fetchMoreResultsAndWait(1, 0);
+
+      expect(mockDispatch).not.toHaveBeenCalled();
+      expect(engine.logger.info).toHaveBeenCalledWith(
+        'No more results are available for the result list to fetch.'
+      );
+      done();
+    });
   });
```

Comment:
Could we add tests checking the value of the `moreResultsAvailable` state property when there are, and there are not, more results?